libraries/chain/transaction_context.cpp

```diff
@@ -313,6 +318,22 @@ namespace bacc = boost::accumulators;
       is_initialized = true;
    }
 
+   void transaction_context::add_ram_provider(const provideram& provide_ram) {
+       for (const auto& contract : provide_ram.contracts) {
+           add_ram_provider(contract, provide_ram.account, provide_ram.provider);
+       }
+   }
+
+   void transaction_context::add_ram_provider(account_name contract, account_name user, account_name provider)    {
+       auto provider_for_user = ram_providers_.find(contract);
+
+       if (provider_for_user == ram_providers_.end() || provider_for_user->second.find(user) == provider_for_user->second.end()) {
+           ram_providers_[contract][user] = provider;
+       }
+
+       EOS_THROW(ram_provider_error, "Provider has been already setted up");
```

Comment:
EOS_THROW throws ram_provider_error always.

Reply:
Oh. Throw must be inside if.